.github/workflows/mobile.yml

```diff
@@ -86,11 +86,42 @@ jobs:
         working-directory: ./my_project/mobile
         run: npm install
 
+      - name: Set Variables
+        id: set-vars
+        run: |
+          echo "backend_url=${{ github.event.deployment_status.environment_url }}" >> $GITHUB_ENV
+          echo "PR_NUMBER=$(echo ${{ github.event.deployment_status.environment_url }} | grep -oP '(?<=pr-)\d+')" >> $GITHUB_ENV
+          echo "REPO_NAME=$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]' | sed 's/[^a-z0-9._-]//g')" >> $GITHUB_ENV
+          echo "EXPO_CHANNEL=$(echo ${{ github.repository }}-$(echo ${{ github.event.deployment_status.environment_url }} | grep -oP '(?<=pr-)\d+') | tr '[:upper:]' '[:lower:]' | sed 's/[^a-z0-9._-]//g')" >> $GITHUB_ENV
       - name: 🚀 Publish preview
         working-directory: ./my_project/mobile
         run: |
-          eas update --branch="${{ github.event.deployment_status.environment_url }}" --non-interactive --auto
-          echo "${{ env.BACKEND_SERVER_URL }} and ${{ github.event.deployment_status.environment_url }}"
+          eas update --branch=$EXPO_CHANNEL --non-interactive --auto
         env:
           EXPO_PUBLIC_BACKEND_SERVER_URL: "${{ github.event.deployment_status.environment_url }}"
-          EXPO_PUBLIC_ROLLBAR_ACCESS_TOKEN: "1a19e5da05b2435b802d5a81aba2bbd7"
\ No newline at end of file
+          EXPO_PUBLIC_ROLLBAR_ACCESS_TOKEN: "1a19e5da05b2435b802d5a81aba2bbd7"
+      
+      - name: Update EAS Config
+        working-directory: ./my_project/mobile
+        run: |
+            jq --arg pr_number "$PR_NUMBER" --arg expo_channel "$EXPO_CHANNEL" '.build["review_" + $pr_number] = {
+              "developmentClient": true,
+              "distribution": "internal",
+              "channel": $expo_channel,
```

Comment:
One thought I had was, if we remove the channel here will we be able to use the statically defined config instead of dynamically creating a new entry?
The reason we are adding this entry is so that there will be a unique build for this PR that does not affect other PR's. However since this one is built on the fly in order to test it the user will have to: 
Register their device, and update the cert to and then rebuild. 

1. eas device:create 
2. eas credentials ---> review_pr-#
3. re build